codec-http2/src/main/java/io/netty/handler/codec/http2/Http2FrameCodecBuilder.java

```diff
@@ -0,0 +1,177 @@
+        return super.gracefulShutdownTimeoutMillis();
+    }
+
+    @Override
+    public Http2FrameCodecBuilder gracefulShutdownTimeoutMillis(long gracefulShutdownTimeoutMillis) {
+        return super.gracefulShutdownTimeoutMillis(gracefulShutdownTimeoutMillis);
+    }
+
+    @Override
+    public boolean isServer() {
+        return super.isServer();
+    }
+
+    @Override
+    public int maxReservedStreams() {
+        return super.maxReservedStreams();
+    }
+
+    @Override
+    public Http2FrameCodecBuilder maxReservedStreams(int maxReservedStreams) {
+        return super.maxReservedStreams(maxReservedStreams);
+    }
+
+    @Override
+    public boolean isValidateHeaders() {
+        return super.isValidateHeaders();
+    }
+
+    @Override
+    public Http2FrameCodecBuilder validateHeaders(boolean validateHeaders) {
+        return super.validateHeaders(validateHeaders);
+    }
+
+    @Override
+    public Http2FrameLogger frameLogger() {
+        return super.frameLogger();
+    }
+
+    @Override
+    public Http2FrameCodecBuilder frameLogger(Http2FrameLogger frameLogger) {
+        return super.frameLogger(frameLogger);
+    }
+
+    @Override
+    public boolean encoderEnforceMaxConcurrentStreams() {
+        return super.encoderEnforceMaxConcurrentStreams();
+    }
+
+    @Override
+    public Http2FrameCodecBuilder encoderEnforceMaxConcurrentStreams(boolean encoderEnforceMaxConcurrentStreams) {
+        return super.encoderEnforceMaxConcurrentStreams(encoderEnforceMaxConcurrentStreams);
+    }
+
+    @Override
+    public Http2HeadersEncoder.SensitivityDetector headerSensitivityDetector() {
+        return super.headerSensitivityDetector();
+    }
+
+    @Override
+    public Http2FrameCodecBuilder headerSensitivityDetector(
+            Http2HeadersEncoder.SensitivityDetector headerSensitivityDetector) {
+        return super.headerSensitivityDetector(headerSensitivityDetector);
+    }
+
+    @Override
+    public Http2FrameCodecBuilder encoderIgnoreMaxHeaderListSize(boolean ignoreMaxHeaderListSize) {
+        return super.encoderIgnoreMaxHeaderListSize(ignoreMaxHeaderListSize);
+    }
+
+    @Override
+    public Http2FrameCodecBuilder initialHuffmanDecodeCapacity(int initialHuffmanDecodeCapacity) {
+        return super.initialHuffmanDecodeCapacity(initialHuffmanDecodeCapacity);
+    }
+
+    /**
+     * Build a {@link Http2FrameCodec} object.
+     */
+    @Override
+    public Http2FrameCodec build() {
+        Http2FrameWriter frameWriter = this.frameWriter;
+        if (frameWriter != null) {
+            // This is to support our tests and will never be executed by the user as frameWriter(...)
+            // is package-private.
+            DefaultHttp2Connection connection = new DefaultHttp2Connection(isServer(), maxReservedStreams());
+            Long maxHeaderListSize = initialSettings().maxHeaderListSize();
+            Http2FrameReader frameReader = new DefaultHttp2FrameReader(maxHeaderListSize == null ?
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Close this "DefaultHttp2FrameReader". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2095)